app/javascript/lib/scp-api.jsx

```diff
@@ -791,18 +795,34 @@ export default async function scpApi(
 
   const perfTimeStart = performance.now()
 
-  const response = await fetch(url, init).catch(error => error)
-
-  // Milliseconds taken to fetch data from API
-  // Suboptimal, but retained until at least Q4 2021 for continuity.
-  // Use `perfTime:full` for closest measure of user-perceived duration.
-  const legacyBackendTime = performance.now() - perfTimeStart
+  const isServiceWorkerCacheEnabled = getSCPContext().isServiceWorkerCacheEnabled
 
   const perfTimes = {
     url,
-    legacyBackend: legacyBackendTime
+    serviceWorkerCacheEnabled: isServiceWorkerCacheEnabled
   }
 
+  let response
+  let isServiceWorkerCacheHit = false
+  let legacyBackendTime
+  if (isServiceWorkerCacheEnabled && init.method === 'GET') {
+    perfTimes.requestStart = perfTimeStart
+    const fetchSWCacheResult = await fetchServiceWorkerCache(url, init)
+    response = fetchSWCacheResult[0]
+    isServiceWorkerCacheHit = fetchSWCacheResult[1]
+    legacyBackendTime = STEP_NOT_NEEDED
+  } else {
+    response = await fetch(url, init).catch(error => error)
+
+    // Milliseconds taken to fetch data from API
+    // Suboptimal, but retained until at least Q4 2021 for continuity.
+    // Use `perfTime:full` for closest measure of user-perceived duration.
+    legacyBackendTime = performance.now() - perfTimeStart
+  }
```

Comment:
Let's keep as much of the cache and perftime logic as we can in `service-worker-cache.js`. 
```suggestion
   if (isServiceWorkerCacheEnabled) {
      response = await fetchServiceWorkerCache(url, init, perfTimes)
   }
   if (!response) {
      // fetch normally
   }
```

Reply:
I began this refactor, then realized it'd be more complex and unduly couple the SW cache to SCP API.  But your suggestion did help me realize a subtle bug in backend time accounting when this cache is enabled, which I fixed.